stdlib/Printf/src/Printf.jl

```diff
@@ -262,11 +262,17 @@ end
 end
 
 # integers
+toint(x) = x
+toint(x::Rational) = Integer(x)
+toint(x::AbstractFloat) = x > typemax(Int128) ?
+                            BigInt(round(x)) : x > typemax(Int64) ?
+                            Int128(round(x)) : Int64(round(x))
```

Comment:
Making this so type-unstable seems bad for both the compiler and runtime performance, as well as making it more complex. What did we do before that this wasn't a problem?

Reply:
It looks like we just called the grisu equivalent of `Ryu.writefixed` with precision `0`. Let me play around and see if we can do that as well.

Reply:
Ok, much faster to do that approach: https://github.com/JuliaLang/julia/pull/37601